File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:        
        if 0 <= len(nums) <= 1:
            return []

        seen = {}
        
        for i, num in enumerate(nums):
            complement = target - nums[i]
            if complement in seen:
                return [seen[complement], i]
            seen[num] = i

File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> frequency;

        for (int num : nums) {
            frequency[num]++;
            if (frequency[num] > 1) {
                return true;   
            }
        }

        return false; 
    }
};
